BaseClasses.py

```diff
@@ -969,6 +969,28 @@ def add_locations(self, locations: Dict[str, Optional[int]],
         for location, address in locations.items():
             self.locations.append(location_type(self.player, location, address, self))
 
+    def add_event(self, location_name: str, item_name: Optional[str] = None,
+                  rule: Optional[Callable[[CollectionState], bool]] = None) -> Item:
+        """
+        Adds an event location/item pair to a region.
```

Comment:
Part of me feels "lock" or "locked" should be somewhere in this sentence, but all the wordings I can think of seem more confusing than it's worth.

Reply:
Hmm, do you think it's necessary? I guess it'd be nice in the sense that it introduces the idea of a locked location, I don't think it is necessary to understand what this function does though